frontend/components/panels/BaseClassPanel.js

```diff
@@ -319,6 +320,24 @@ class BaseClassPanel extends React.Component {
       </div>
     );
   }
+
+  // Just used for testing. Not used anywhere in the frontend.
+  render() {
```

Comment:
I'd rather not have this code here, but I see why this needs to be here. Perhaps a future PR?

Reply:
Any ideas on how we could test this file without this code? I tried testing it by directly instantiating it (`let instance = new BaseClassParser(aClass)` - which dosen't require a render method), but when I did that it didn't work because any calls to `this.setState` inside BaseClassParser were asynchronous and didn't take effect until after the test finished. With the built in jest rendering for components, the calls to `this.setState` inside BaseClass parser were synchronous which allowed us to actually test their effects.  

I'm down to just leave this here and call it a day haha - I'd rather spend time working on new feature and fixing bugs and stuff :)

Lmk what your thoughts are!

Reply:
Have you tried using Promises? That could solve this issue; I may take a look at it later, but this isn't a large issue to begin with~

Reply:
If we used promises I think we would have to modify every function that called `this.setState` to return a promise that resolved when `this.setState` finished. I'll merge this PR for now and we can come back to it later!